### PR TITLE
Sphinx DFT code

### PR DESCRIPTION
**Pull request by jan-janssen**
@conda-forge/help-c-cpp 

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/sphinxdft/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# MKL patch - requires autoconf
+mkdir -p src/playground
+touch src/playground/Makefile.am
+autoreconf -vif
+rm -rf src/playground
+
+# Setup
+export CXX=$GXX
```
Comment (chrisburr) [inline recipes/sphinxdft/build.sh]: Is this line needed?
```suggestion
```
(I think it's the same on Linux and is unset on macOS)

**Comment by jan-janssen** [comment]
License: 

```
=============================================================================
               ACADEMIC LICENSE OF THE S/PHI/nX package
=============================================================================


A. DEFINITIONS

   1. "The S/PHI/nX package" is:
         the suite of software but also all other files such as documentations,
         scripts, and ancillary support material distributed as a single 
         entity. This License covers the entirety of said data.

   2. "The Distributors" are:
         Sixten Boeck
         Christoph Freysoldt
         Joerg Neugebauer

   3. "The copyright holder" is:
         Max-Planck-Institute fuer Eisenforschung, GmbH



B. NO WARRANTY

   1. There is no warranty for any part of the S/PHI/nX package. 

   2. The S/PHI/nX package is provided by the Distributors "as-is" without
      warranty of any kind. 

   3. In no event shall the Distributors or the Copyright Holder be liable 
      for any direct, indirect, incidental, special, exemplary, or
      consequential damages (including, but not limited to, procurement of
      substitute goods or services; loss of use, data, or profits; or business
      interruption) however caused and on any theory of liability, whether in
      contract, strict liability, or tort (including negligence or otherwise)
      arising in any way out of the use of this software, even if advised of
      the possibility of such damage. 

   4. The entire risk as to the quality or performance of the program, or any
      action the Licensee may take based on the output of any part of the 
      S/PHI/nX package, is with the Licensee.


C. LIMITATION OF USAGE

   1. The use of the S/PHI/nX package under the present License is restricted
      solely to academic institutions (including universities and non-military
      public research laboratories) and to academic purposes, i.e., leading to
      open publication of results, with neither withhold of information nor
      intentional delay in publication. 

   2. Notwithstanding these general definitions, the Distributors reserve the
      exclusive right to determine what purposes and institutions are elegible
      for using the S/PHI/nX package under this License, and to withhold the
      right to use the S/PHI/nX package where they are not satisfied that 
      either condition is fulfilled.

   3. The Licensee is hereby provided with a non-exclusive, royalty-free right
      to use any part of the S/PHI/nX package, and to display, publish, perform,
      or otherwise use results obtained by use of the S/PHI/nX package in any
      form, subject to the restriction that in any paper or other academic
      publication containing results wholly or partially derived from the
      results of use of the S/PHI/nX package, the following paper must be
      cited in the normal manner:

        The object-oriented DFT program library S/PHI/nX
        S. Boeck, Ch. Freysoldt, A. Dick, L. Ismer, J. Neugebauer,
        Comp. Phys. Comm. 182, 543 (2011)

   4. New features which will be implemented in future releases of the S/PHI/nX
      package will be documented in additional feature papers. If these new 
      features have been used for a academic publication the corresponding
      feature papers have to be cited along with the original S/PHI/nX paper
      mentioned in (C.3). The list of additional feature papers is updated
      by the Copyright Holder.

   5. Every work derived from any part of the S/PHI/nX package must visibly
      pay credit to the S/PHI/nX package and must retain the above citation
      clause (C.3) to the licensee of the derived work.

   6. All advertising materials mentioning features or use of the S/PHI/nX
      package or its derived works must display the following 
      acknowledgement:

        http://www.sphinxlib.de

   7. Neither the name of the Max-Planck-Institut fuer Eisenforschung GmbH nor
      the names of the Distributors may be used to endorse or promote products
      derived from the S/PHI/nX package without specific prior written
      permission.

D. COMMERCIAL USAGE

  1. The present license excludes any commercial usage.

  2. Commercial usage implies all other uses than defined in C1, notably
     a) use in companies for profit,
     b) asking or accepting payment from third-party sources for results
        obtained by the S/PHI/nX package
     c) commercially distributing the S/PHI/nX package as a whole or in parts
     d) or commercially distributing work derived from the S/PHI/nX package.

     Clarification: 
     Third-party funded research projects carried out in academic institutions
     are usually not considered commercial usage if the results are published
     according to common academic practice.

  3. A commercial distribution of parts of the S/PHI/nX package may be allowed
     if the development of the derived work is sufficiently substential.
     Commercial distribution of the S/PHI/nX package or parts of it
     requires the written permission of the 
     Max-Planck-Institut fuer Eisenforschung GmbH.

  4. An individual license for commercial use may be obtained from the 
     Copyright Holder.

E. REDISTRIBUTION

   1. Redistributions of any part of the S/PHI/nX package must retain the 
      present license terms including the advertising clauses (C.3 - C.5).

   2. Everyone is permitted to copy and distribute verbatim copies of this
      License document, but changes are not permitted.

   3. Modifications of these license terms for any derived work may be allowed
      and require the written permission of the 
      Max-Planck-Institut fuer Eisenforschung GmbH.

F. SUPPORT

   1. The present License does not imply support services of any kind to the
      Licensee by the Distributors, including, but not limited to, 
      installation, compilation and use of the software in the package.


G. ALL OTHER RIGHTS

   1. All other rights not explicitly mentioned in this document are reserved.
  
   2. In particular, the Distributors reserve all rights to resolve any and all
      conflicts arising in the development and distribution policies of the 
      S/PHI/nX package.

   3. Without a valid License, a prospective Licensee does not have the right
      to use or modify the S/PHI/nX package.

H. BINARIES

   1. The Distributors may provide compiled versions (binaries) of the SPHInX 
      package. The use of binaries is covered by this License with additional
      restrictions as specified below.

   2. 3rd-party material contained in binary versions may only be used in 
      conjunction with the SPHInX binaries provided. It may not be used for
      other purposes.

   3. The Distributors of the SPHInX package give no warranty whatsoever (as
      described in paragraph B of this License) for 3rd-party material included
      in binary packages. Likewise, distributors of 3rd-party material are held
      free from any liability arising from the use of any material provided
      in the binary package. The full risk is with the Licensee.

   4. Binary packages must not be used if the potential Licensee is prohibited
      from using any contained material by law or by the Distributors of
      3rd-party material. If such a conflict occurs, any material obtained and
      copies thereof must be destroyed. The Licensee will indemnify, hold
      harmless, and defend the Distributors from and against any claims or
      lawsuits, including attorney's fees, that arise or result from prohibited
      use of 3rd-party material.

   4. Binaries must not be modified, reverse engineered, decompiled, or 
      disassembled.

   5. Redistribution of binary versions is prohibited.
```

**Comment by jan-janssen** [comment]
@chrisburr I am part of the Max-Planck-Institut fuer Eisenforschung GmbH https://www.mpie.de/person/42524/2768816 and I got the confirmation from Christoph Freysoldt and Joerg Neugebauer to integrate the code in conda-forge. @bgruening can confirm this.

File: recipes/sphinxdft/build.sh (hunk 1 above)
Comment (jan-janssen) [reply recipes/sphinxdft/build.sh]: The compilation is a bit tricky, the sphinx `configure` script is not able to recognise: 

```
CXX=$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++
```

as g++ but it works fine with:

```
GXX=$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-g++ 
```

Therefore I had to overwrite the variable.

Comment (chrisburr) [reply recipes/sphinxdft/build.sh]: In that case can you leave include a brief comment in the script.

File: recipes/sphinxdft/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{% set name = "sphinxdft" %}
+{% set version = "2.6.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://sxrepo.mpie.de/attachments/download/42/sphinx-2.6.tar.xz
```
[the remark this answers is not in the document]
Comment (jan-janssen) [reply recipes/sphinxdft/meta.yaml]: But the  version is 2.6.0 while the package is stored under 2.6 ...

File: recipes/sphinxdft/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{% set name = "sphinxdft" %}
+{% set version = "2.6.0" %}
```
[the remark this answers is not in the document]
Comment (jan-janssen) [reply recipes/sphinxdft/meta.yaml]: We plan to change the upstream version to the three digit format in future.

File: recipes/sphinxdft/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisburr) [reply recipes/sphinxdft/meta.yaml]: In that case you can use jinja2 trickery such as:
```suggestion
  url: https://sxrepo.mpie.de/attachments/download/42/sphinx-{{ version[:-2] if version.endswith('.0') else version }}.tar.xz
```

**Comment by jan-janssen** [comment]
@chrisburr I included all your recommendations and we can definitely write an email to NumFOCUS to confirm the permission to distribute the sphinx DFT code.

**Comment by jan-janssen** [comment]
After some more discussions with @bgruening I realise that it is most likely easier if we change the license of the sphinx DFT code to an open source license - to be on the save side. I am going to start this process internally, but it might take a few more days to get the corresponding approvals.

**Comment by jan-janssen** [comment]
With the release of version 2.6.1 we fixed the version naming scheme and changed the license to Apache.

**Comment by bgruening** [comment]
Wow, that was fast! You and your team rocks!

**Pull request merged**